mslib/mscolab/server.py

```diff
@@ -80,31 +80,43 @@ class mscolab_auth:
 
 # setup idp login config
 if mscolab_settings.USE_SAML2:
+    # saml_2_backend yaml config
     with open(f"{mscolab_settings.MSCOLAB_SSO_DIR}/mss_saml2_backend.yaml", encoding="utf-8") as fobj:
         yaml_data = yaml.safe_load(fobj)
 
     # go through configured IDPs and set conf file paths for particular files
     for configured_idp in mscolab_settings.CONFIGURED_IDPS:
         # set CRTs and metadata paths for the localhost_test_idp
-        if 'localhost_test_idp' in configured_idp['idp_identity_name']:
+        if 'localhost_test_idp' == configured_idp['idp_identity_name']:
             yaml_data["config"]["localhost_test_idp"]["key_file"] = \
                 f'{mscolab_settings.MSCOLAB_SSO_DIR}/key_mscolab.key'
             yaml_data["config"]["localhost_test_idp"]["cert_file"] = \
                 f'{mscolab_settings.MSCOLAB_SSO_DIR}/crt_mscolab.crt'
             yaml_data["config"]["localhost_test_idp"]["metadata"]["local"][0] = \
                 f'{mscolab_settings.MSCOLAB_SSO_DIR}/idp.xml'
 
-    if not os.path.exists(yaml_data["config"]["localhost_test_idp"]["metadata"]["local"][0]):
-        yaml_data["config"]["localhost_test_idp"]["metadata"]["local"] = []
-        warnings.warn("idp.xml file does not exists ! Ignore this warning when you initializeing metadata.")
+            # configuration localhost_test_idp Saml2Client
+            try:
+                if not os.path.exists(yaml_data["config"]["localhost_test_idp"]["metadata"]["local"][0]):
+                    yaml_data["config"]["localhost_test_idp"]["metadata"]["local"] = []
+                    warnings.warn("idp.xml file does not exists ! Ignore this warning when you initializeing metadata.")
+
+                localhost_test_idp = SPConfig().load(yaml_data["config"]["localhost_test_idp"])
+                sp_localhost_test_idp = Saml2Client(localhost_test_idp)
+
+            except SAMLError:
+                warnings.warn("Invalid Saml2Client Config with localhost_test_idp ! Please configure with valid CRTs\
+                              /metadata and try again.")
+                sys.exit()
+
+        # if multiple IdPs exists, development should need to implement accordingly below
```

Comment:
keep in mind that this file can't be changed when the software gets installed.  When someone would edit the file on the next release all that changes are gone.

Reply:
Thank you for figuring it out. I have updated the PR accordingly. : )